src/main/src/test/resources/org/geoserver/config/external_with_params_and_fragment.sld

```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
+<StyledLayerDescriptor version="1.0.0" xmlns="http://www.opengis.net/sld" xmlns:ogc="http://www.opengis.net/ogc"
+  xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://www.opengis.net/sld http://schemas.opengis.net/sld/1.0.0/StyledLayerDescriptor.xsd">
+  <NamedLayer>
+    <Name>external</Name>
+    <UserStyle>
+      <Name>external</Name> 
+      <Title>An external graphic</Title>
+      <Abstract>A sample point symbolizer with an external graphic with a URL containing both query parameters and a # fragment</Abstract>
+      <FeatureTypeStyle>
+        <Rule>
+          <Title>Red flag</Title>
+          <PointSymbolizer>
+            <Graphic>
+              <ExternalGraphic>
+                <OnlineResource xlink:type="simple" xlink:href="icon.png?param1=1#textAfterHash" />
```

Comment:
Looks good, I assume you got this example from the sprite case. The wkt.properties#symbol from the docs is the one we are trying to fix.

Suggestion only no need to change.